Spigot-Server-Patches/0755-Fix-plugin-loggers-on-server-shutdown.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Jos=C3=A9=20Miguel=20Moreno?= <josemmo@pm.me>
+Date: Sat, 5 Jun 2021 13:45:15 +0200
+Subject: [PATCH] Fix plugin loggers on server shutdown
+
+
+diff --git a/src/main/java/io/papermc/paper/log/CustomLogManager.java b/src/main/java/io/papermc/paper/log/CustomLogManager.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..c1d3bac79bb8b4796c013ff4472f75dcd79602dc
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/log/CustomLogManager.java
+@@ -0,0 +1,26 @@
++package io.papermc.paper.log;
++
++import java.util.logging.LogManager;
++
++public class CustomLogManager extends LogManager {
++    private static CustomLogManager instance;
++
++    public CustomLogManager() {
++        instance = this;
++    }
++
++    @Override
++    public void reset() {
++        // Ignore calls to this method
++    }
++
++    private void superReset() {
++        super.reset();
++    }
++
++    public static void forceReset() {
++        if (instance != null) {
++            instance.superReset();
++        }
++    }
++}
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index bf80e870e6a2a6fe1d4ae1bea355bcd7a0735d3b..9bbad437983d406ceb285ad2a431a653b270a73b 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -14,6 +14,7 @@ import io.netty.buffer.ByteBufOutputStream;
+ import io.netty.buffer.Unpooled;
+ import io.papermc.paper.adventure.PaperAdventure; // Paper
+ import io.papermc.paper.event.entity.EntityMoveEvent;
++import io.papermc.paper.log.CustomLogManager;
```

Comment:
Don't import here, use FQN below

Reply:
I don't know whether that's possible. When using FQN instead of an import I get the following error trying to package the Paper-Server module:
```
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project paper: Compilation failure
/home/josemmo/PaperMC/Paper-Server/src/main/java/net/minecraft/server/MinecraftServer.java:[973,11] cannot find symbol
  symbol:   variable papermc
  location: class net.minecraft.server.io
```

Besides, all other `io.papermc.*` classes used in `net.minecraft.server.MinecraftServer` have an import statement.

Reply:
Fully qualified in 1.16.5 won't work because of the ever loving `io` class, this will work fine in 1.17

Reply:
Is there any documentation on how to build Paper 1.17? I'm trying to rebase this branch and can't even apply patches to master at commit 98fa13609982b3ee19928efed96f4d23735ea100:

```
> Task :patchCraftBukkitPatches FAILED
error: patch failed: .gradle/caches/paperweight/taskCache/patchCraftBukkitPatches.zip-1711287088/net/minecraft/world/entity/ai/behavior/BehaviorWorkComposter.patch:1
error: .gradle/caches/paperweight/taskCache/patchCraftBukkitPatches.zip-1711287088/net/minecraft/world/entity/ai/behavior/BehaviorWorkComposter.patch: patch does not apply        
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':patchCraftBukkitPatches'.
> io.papermc.paperweight.PaperweightException: Command finished with 1 exit code: git -c commit.gpgsign=false -c core.safecrlf=false apply C:\Users\josemmo\Desktop\PaperMC\build-data\craftbukkit-patch-patches\BehaviorWorkComposter.patch.patch
```

The failing git command prints the following error message:
```
error: net/minecraft/world/entity/ai/behavior/BehaviorWorkComposter.patch: No such file or directory
```